### ISSUE TITLE
Preprocess of cutting-off the cloth area

### ISSUE THREAD

**Issue by summertight** [Zielon/GEM#3]
Dear Zielonka,

Thank you for your great work and support.

I’m reaching out with a question regarding the performance of BiSeNet. I followed the instructions provided in the README.md and used the [BiSeNet for face parsing](https://github.com/zllrunning/face-parsing.PyTorch) repository as recommended, to segment out the cloth region. However, the results seem to be less robust than expected, as shown in the attached image.

For reference, I’ve modified the render_mask.py script as follows to incorporate the cloth mask:

All the best, 
Jaeseong Lee

```
cloth_mask = face_parser.forward(image[None])[-1].squeeze(0).to(torch.float32)
neck_mask = (rasterize_map(Renderer.to_cameras(single), mesh))
full_mask = (neck_mask + cloth_mask).clamp(0,1)
rendering = (1-full_mask) * alpha
rendering = torch.clamp(rendering, 0, 1) 
```
![Image](https://github.com/user-attachments/assets/cd5f7a94-b305-47a2-a5f8-4636a53a17ee)

**Comment by Zielon** [comment]
Hello,

You could either **use only** face parsing to remove the garment. Or try to change the order of operations:

alpha_mask - from background matting
cloth_mask - from BiSeNet
neck_mask - from mesh

1) Remove the cloth using BiSeNet image = image * cloth_mask
2) Inflate the mesh and render neck_mask (1 or 2 cm?)
3) Use mask = (1 - neck_mask) * alpha_mask
4) image * mask

I haven't tested it. But I remember experimenting with something like this.

Best

**Comment by summertight** [comment]
Thanks.
